@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -320,4 +378,73 @@ export class SimpleTileGeometryLoader implements TileGeometryLoader {
             tile.dataSource.requestUpdate();
         }, 0);
     }
+
+    /**
+     * Compare enabled and disabled geometry kinds with currently set.
+     *
+     * Method compares input sets with recently used geometry kinds in performance wise
+     * manner, taking special care of undefined and zero size sets.
+     *
+     * @param enabledKinds Set of geometry kinds to be displayed or undefined.
+     * @param disabledKinds Set of geometry kinds that won't be rendered.
+     * @return `true` only if sets are logically equal, meaning that undefined and empty sets
+     * may result in same geometry (techniques kind) beeing rendered.
+     */
+    private compareGeometryKinds(
```

Comment:
How is this related to the task queue.
I think nobody is using geometry kind filtering so I would rather drop that feature than adding even more code b/c of a feature nobody uses.

Reply:
I would agree, but this was introduced for Phased Loading and actually never supported in SimpleLoader! This provides support for it.
In simple words, I have a bad habit that whenever I touch something I am trying to at least comply with the interfaces somebody have prepared.
If we want to remove this we will need to drop PhasedLoading too, I guess and as far as I know there is separate ticket for this.

Reply:
Just a side note, maybe instead of trying to support kinds update, we could simply leave it as it was but simply add:
```
assert(enabledKinds == undefined || enabledKinds.size() === 0);
assert(disabledKinds == undefined || disabledKinds.size() === 0);
```
but this would assume that no one is using it... I heard that this is not purely true...

Reply:
I like your second proposal to avoid adding more dead code.
Right now we anyways crash or assert when phased loading is enabled. So we should just ignore the `enablePhaseLoading` flag and never enable it.